### PR TITLE
added few more plugin configurations entity_id, entity_count, entity_lis...

### PR DESCRIPTION
**Pull request by marcogiacomassi**
In my project I needed a way to easily set entity ID and list name to a custom name.
For example:

``` javascript
{
    "products": [
        {
            "pid": 310,
            "tnid": "309",
            "title": "product name",
            "status": "1",
            "description": "product descr"
                }
        ]
}
```

I think this could be done from plugin configuration without the needs to extend RestfulEntityBaseNode class.

These are the new parameters added to plugin config

``` php
'entity_id' => 'pid',
'entity_list_name' => 'products',
'entity_count' => TRUE,
```

This also adds a count of the returned items.

### CODE REVIEW

File: plugins/restful/RestfulEntityBase.php [hunk 1]
```diff
@@ -343,6 +343,11 @@ public function __construct($plugin, \RestfulAuthenticationManager $auth_manager
     $this->bundle = $plugin['bundle'];
     $this->authenticationManager = $auth_manager ? $auth_manager : new \RestfulAuthenticationManager();
     $this->cacheController = $cache_controller ? $cache_controller : $this->newCacheObject();
+    
+    $this->entityId = !empty($plugin['entity_id']) ? $plugin['entity_id'] : 'id';
```
Comment (e0ipso) [inline plugins/restful/RestfulEntityBase.php:347]: I think that this is confusing. Maybe something like `entityIdKey`?

File: plugins/restful/RestfulEntityBase.php [hunk 2]
```diff
@@ -343,6 +343,11 @@ public function __construct($plugin, \RestfulAuthenticationManager $auth_manager
...
+    $this->entityCount = !empty($plugin['entity_count']) ? $plugin['entity_count'] : FALSE;
```
Comment (e0ipso) [inline plugins/restful/RestfulEntityBase.php:348]: How does this account pagination? Is this the total number? Is this the number of items per page?

File: plugins/restful/RestfulEntityBase.php [hunk 3]
```diff
@@ -538,14 +543,18 @@ public function getList() {
...
+    $return = array($this->entityListName => array());
 
     $this->getListAddHateoas($return, $ids);
 
     foreach ($ids as $id) {
-      $return['list'][] = $this->viewEntity($id);
+      $return[$this->entityListName][] = $this->viewEntity($id);
+    }
+    
+    // add count if requested in plugin config
```
Comment (e0ipso) [inline plugins/restful/RestfulEntityBase.php]: `// add count if requested in plugin config`
to
`// Add count if requested in plugin config.`

File: plugins/restful/RestfulEntityBase.php (hunk 1 above)
Comment (e0ipso) [inline plugins/restful/RestfulEntityBase.php:347]: Actually I'd go with whatever the name is in core/entity API module.

File: plugins/restful/RestfulEntityBase.php (hunk 2 above)
Comment (marcogiacomassi) [inline plugins/restful/RestfulEntityBase.php:348]: You are right, this should be the the total number

File: plugins/restful/RestfulEntityBase.php (hunk 1 above)
Comment (marcogiacomassi) [inline plugins/restful/RestfulEntityBase.php:347]: I think entity_id_key and entityIdKey are good! In core/entity API module are named entity keys

**Comment by amitaibu** [comment]
Sorry, in vacation. I'll review once I'm back :)

**Comment by amitaibu** [comment]
Thank you for the contribution, but I'm not a fan of this PR. Those changes can be pretty easily be done by overriding some methods. 
I'd like to keep the plugin configuration as much as possible out of the final data structure.

If you feel I'm missing something, please re-open.

**Pull request closed without merging**